state-sync/src/coordinator.rs

```diff
@@ -400,194 +318,265 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
         );
 
         self.sync_state_with_local_storage()?;
-        ensure!(
-            self.is_initialized(),
-            "[state sync] Sync request but initialization is not complete!"
-        );
-        if target_version == local_li_version {
-            return Self::send_sync_req_callback(request, Ok(()));
+        if !self.is_initialized() {
+            return Err(UninitializedError(
+                "Unable to process sync request message!".into(),
+            ));
         }
 
+        if target_version == local_li_version {
+            return Ok(Self::send_sync_req_callback(request, Ok(()))?);
+        }
         if target_version < local_li_version {
             Self::send_sync_req_callback(request, Err(format_err!("Sync request to old version")))?;
-            bail!(
-                "[state sync] Sync request for version {} < known version {}",
-                target_version,
-                local_li_version,
-            );
+            return Err(OldSyncRequestVersion(target_version, local_li_version));
         }
 
         self.sync_request = Some(request);
-        self.send_chunk_request(
+        Ok(self.send_chunk_request(
             self.local_state.synced_version(),
             self.local_state.trusted_epoch(),
-        )
+        )?)
```

Comment:
Yeah, this is for error conversion. Ideally, if everything was using our error types, we wouldn't need this, but because `send_chunk_request` returns an anyhow error, and this method returns an error::Error, we use `?` to do the conversion. This should eventually go away though 😄